lib/src/store.rs

```diff
@@ -124,9 +136,20 @@ impl Store {
         Ok(data)
     }
 
-    pub fn write_commit(self: &Arc<Self>, commit: backend::Commit) -> BackendResult<Commit> {
+    pub fn write_commit(
+        self: &Arc<Self>,
+        commit: backend::Commit,
+        sign_config: SignSettings,
+    ) -> BackendResult<Commit> {
         assert!(!commit.parents.is_empty());
-        let (commit_id, commit) = self.backend.write_commit(commit, None)?;
+
+        let signing_fn = (self.signer.can_sign() && sign_config.should_sign(&commit)).then(|| {
+            let store = self.clone();
+            Box::new(move |data: &_| store.signer.sign(data, sign_config.key.as_deref()))
+                as SigningFn
+        });
```

Comment:
It's convenient for store to have the signer for the `Commit::verification` method

Reply:
> It's convenient for store to have the signer for the `Commit::verification` method

Yeah. I though `Arc<Signer>` could be copied from repo to `Commit`, but found that `backend::Commit` is wrapped by `Store`, not by repo.

> And then the check if the new commit should be signed being here follows from that

Still I think `SignBehavior` is high-level concept that the store shouldn't know.

Reply:
Well, we still have the signer owned by the store because the store wraps commits, but I moved the signing function creation all the way up to the commit builder.

It also makes slightly more sense there because I need to first check the behaviour and then clear out the `secure_sig` field because I decided to have that assert in the backend :)